src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -120,14 +123,40 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         if (this.useSsh && this.sshEnabled && this.sshTemplateUrl) {
             return this.getSshCloneUrl(this.getRepositoryUri()) || this.getRepositoryUri();
         }
-
         if (this.isTeamParticipation) {
             return this.addCredentialsToHttpUrl(this.repositoryUriForTeam(this.getRepositoryUri()), insertPlaceholder);
         }
 
         return this.addCredentialsToHttpUrl(this.getRepositoryUri(), insertPlaceholder);
     }
 
+    loadVcsAccessToken() {
```

Comment:
In the past, the additional request to fetch a token was deemed undesirable (#7486). Is this okay now?

Reply:
tbh I wasn't aware of this PR. I guess one difference is, that https://github.com/ls1intum/Artemis/pull/7486 fetches the whole user object again, whereas here only the token itself is loaded